.github/actions/check-current-workflow-is-latest/action.yml

```diff
@@ -0,0 +1,30 @@
+name: Check current workflow is the latest workflow
+description: Checks whether the current workflow is the most recent workflow.
+
+inputs:
+  current-workflow-id:
+    description: The current workflow run's database id.
+    required: true
+    default: ""
+
+outputs:
+  should-run:
+    description: The latest workflow run's database id.
+    value: ${{ steps.check-current-workflow-equals-latest.outputs.should-run }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Scripts
+      shell: pwsh
+      run: |
+        "${{ github.action_path }}" >> $Env:GITHUB_PATH
+        (Resolve-Path "${{ github.action_path }}/../../../Scripts").Path >> $Env:GITHUB_PATH
+      
+    - name: Check Current Workflow Equals Latest
+      shell: pwsh
+      id: check-current-workflow-equals-latest
+      run: |
+        $latestWorkflowRunNumber=$(gh run list --limit 1 --json databaseId | jq -r '.[0].databaseId')
```

Comment:
`gh run list` lists all workflow runs from the current repo, thus this line is fetching the most recent workflow run for every pull request too. This is not what we need. Only the workflow runs of the current PR are interesting, since other workflow runs are irrelevant to this action.